src/components/views/messages/MImageBody.js

```diff
@@ -235,59 +237,87 @@ export default class extends React.Component {
     _afterComponentWillUnmount() {
     }
 
-    onAction(payload) {
-        if (payload.action === "timeline_resize") {
-            this.fixupHeight();
+    _messageContent(contentUrl, thumbUrl, content) {
+        // The maximum height of the thumbnail as it is rendered as an <img>
+        const maxHeight = Math.min(THUMBNAIL_MAX_HEIGHT, content.info.h);
```

Comment:
I'm still really surprised that we can't just set `max-height: 600px` on the image rather than calculating this in JS...